### PR TITLE
Enable SonarCloud code scanning

### LINKED ISSUE #1175: Enable SonarCloud
**Problem**
We should scan our code with free tools like [SonarCloud](https://sonarcloud.io/) to find vulnerabilities and bugs.

**Solution**
- Add our repository to [SonarCloud](https://sonarcloud.io/) to automatically scan JavaScript code
- Add a task to an existing GitHub workflow to invoke SonarCloud after the Java code is built to manually scan compiled code
- Fix any discovered flaws (as separate tickets)

**Alternatives**

**Additional Context**

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Detailed description**:
- Add SonarCloud code scanning to `Code Scan` GitHub action job
- Bump a few vulnerable dependencies
- Fix a brittle test `CompositeTransactionGeneratorTest`
- Split Java and JavaScript code scan into separate jobs to reduce conditional complexity

**Which issue(s) this PR fixes**:
Fixes #1175 

**Special notes for your reviewer**:
https://sonarcloud.io/dashboard?id=hedera-mirror-node

Couldn't get code coverage to look correct so removed running tests for code scanning for now

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -151,6 +158,7 @@
                         <!-- Jacoco prepare-agent builds some command-line params without -->
                         <!-- which jacoco will not instrument. Hence it is important to add -->
                         <!-- those command-line params here (${argLine} holds those params) -->
+                        <!--suppress UnresolvedMavenProperty -->
```
Comment (Nana-EC) [inline pom.xml:161]: Does this line need to be here or is it a dupe of line 77?

File: pom.xml [hunk 2]
```diff
@@ -73,7 +74,13 @@
         <protobuf.version>3.13.0</protobuf.version>
         <release.version>${project.version}</release.version> <!-- Used to replace release versions in all files -->
         <release.chartVersion>0.9.0-rc1</release.chartVersion>
-        <embedded.testcontainers.version>1.85</embedded.testcontainers.version>
+        <!--suppress UnresolvedMavenProperty -->
+        <sonar.coverage.jacoco.xmlReportPaths>
```
Comment (Nana-EC) [inline pom.xml:78]: nit: better yet a comment here saying "point to directory where sonar can import jacoco calculated coverage report from"

File: pom.xml (hunk 1 above)
Comment (steven-sheehy) [reply pom.xml:161]: It's not a dupe, it suppresses warnings in intellij about the line that follows it.

File: pom.xml (hunk 2 above)
Comment (steven-sheehy) [reply pom.xml:78]: This isn't a descriptive comment, it's a programmatic comment to tell intellij the next line is not an error.

Comment (Nana-EC) [reply pom.xml:78]: I wasn't referring to the suppress in this case, I meant it might be useful to comment on the sonar jacoco relation

Comment (steven-sheehy) [reply pom.xml:78]: I don't see how `sonar.coverage.jacoco.xmlReportPaths` could be unclear? It contains every keyword in your example description.

**Review by Nana-EC — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**